pkg/pool-utils/test/BasePoolController.test.ts

```diff
@@ -0,0 +1,236 @@
+      await expect(poolController.connect(admin).setSwapFeePercentage(NEW_SWAP_FEE)).to.be.revertedWith(
+        'UNINITIALIZED'
+      );
+    });
+
+    it('has no initial metadata', async () => {
+      expect(await poolController.getMetadata()).to.equal('0x');
+    });
+
+    it('owner can set metadata', async () => {
+      const tx = await poolController.connect(admin).updateMetadata(METADATA);
+      expectEvent.inReceipt(await tx.wait(), 'MetadataUpdated', {
+        metadata: METADATA,
+      });
+
+      expect(await poolController.getMetadata()).to.equal(METADATA);
+    });
+
+    it('non-owner cannot set metadata', async () => {
+      await expect(poolController.connect(other).updateMetadata(METADATA)).to.be.revertedWith('CALLER_IS_NOT_OWNER');
+    });
+  });
+
+  context('pool controller is initialized', () => {
+    sharedBeforeEach('initialize pool controller', async () => {
+      await poolController.bindPool(pool.address);
+    });
+
+    it('owner can set the swap fee', async () => {
+      await poolController.connect(admin).setSwapFeePercentage(NEW_SWAP_FEE);
+
+      expect(await pool.getSwapFeePercentage()).to.equal(NEW_SWAP_FEE);
+    });
+
+    it('enforces owner permissions', async () => {
+      await expect(poolController.connect(other).setSwapFeePercentage(NEW_SWAP_FEE)).to.be.revertedWith(
+        'CALLER_IS_NOT_OWNER'
+      );
+    });
+
+    it('can be transferred', async () => {
+      await poolController.transferOwnership(other.address);
+
+      await expect(poolController.connect(admin).setSwapFeePercentage(NEXT_SWAP_FEE)).to.be.revertedWith(
+        'CALLER_IS_NOT_OWNER'
+      );
+
+      await poolController.connect(other).setSwapFeePercentage(NEXT_SWAP_FEE);
+
+      expect(await pool.getSwapFeePercentage()).to.equal(NEXT_SWAP_FEE);
+    });
+
+    it('can renounce ownership', async () => {
+      await poolController.renounceOwnership();
+
+      await expect(poolController.connect(admin).setSwapFeePercentage(NEXT_SWAP_FEE)).to.be.revertedWith(
+        'CALLER_IS_NOT_OWNER'
+      );
+
+      expect(await poolController.owner()).to.equal(ZERO_ADDRESS);
+    });
+
+    it('cannot be paused by owner', async () => {
+      await expect(pool.setPaused(true)).to.be.revertedWith('SENDER_NOT_ALLOWED');
+    });
+
+    context('pause authorized', async () => {
+      sharedBeforeEach('authorize governance to pause', async () => {
+        const action = await actionId(pool.instance, 'setPaused');
+        await vault.grantRole(action, governance.address);
+      });
+
+      it('still cannot be paused by owner', async () => {
+        await expect(pool.setPaused(true)).to.be.revertedWith('SENDER_NOT_ALLOWED');
+      });
+
+      it('can be paused by governance', async () => {
+        await pool.instance.connect(governance).setPaused(true);
+
+        expect(await pool.isPaused()).to.be.true;
+      });
+    });
+
+    describe('Asset Managed pool', () => {
```

Comment:
I don't think we need to have an entirely separate section for asset managers, repeating all pool setup. We could just set some of the pool's tokens to have asset managers, and then test the setting the config works for them in an owner-only manner.